addons/mail/static/tests/messaging_menu/messaging_menu_tests.js

```diff
@@ -175,6 +180,67 @@ QUnit.test("rendering with PWA installation request", async (assert) => {
     assert.verifySteps(["show prompt"]);
 });
 
+QUnit.test("installation of the PWA request can be dismissed", async (assert) => {
+    const target = getFixture();
+
+    patchWithCleanup(browser, {
+        BeforeInstallPromptEvent: () => {},
+    });
+    patchWithCleanup(browser.localStorage, {
+        getItem(key) {
+            if (key === "pwa.installationState") {
+                assert.step("getItem " + key);
+                // in this test, installation has not yet proceeded
+                return null;
+            }
+            return super.getItem(key);
+        },
+        setItem(key, value) {
+            if (key === "pwa.installationState") {
+                assert.step("installationState value:  " + value);
+            }
+            return super.setItem(key, value);
+        },
+    });
+
+    const { env } = await start();
+    // This event must be triggered to initialize the installPrompt service properly
+    // as if it was run by a browser supporting PWA (never triggered in a test otherwise).
+    window.dispatchEvent(new CustomEvent("beforeinstallprompt"));
+
+    patchWithCleanup(env.services.installPrompt, {
+        show() {
+            assert.step("show prompt should not be triggered");
+        },
+    });
+    assert.verifySteps(["getItem pwa.installationState"]);
+
```

Comment:
Nitpicking but empty newline adds many irrelevant LOCs in test.
This doesn't seem to be a problem at a glance, but in practice these files are big, and newlines separator helps shaping each tests. The empty newlines disturbs reading tests at a glance, which results to more effort and time wasted.

And also it's not great to scroll to read the whole test...

Reply:
I disagree with you, as each individual test is painful to read in this condensed file